### DISCUSSION TITLE
Why DialCode is mandatory ?

### DISCUSSION THREAD

**Discussion by karthik-tarento** [Sunbird-Knowlg/Community#71] [General Discussions]
Hello, 
We are trying to create course without dialCode. And also the meta attribute "dialcodeRequired" is set to "No".
However, the post-publish-processor Flink job is trying to reserve the dial code for the course and it fails to do so. Because of the failure, the "post-publish-processor" keep crashing and other required activities like auto batch creation is not getting successful for further courses when publishing. 

How to make the dial code creation as optional? Is there any configuration to specify default dial codes?

@vinukumar-vs @Jayaprakash8887 - please suggest on this.

Regards,
Karthik

**Comment by Jayaprakash8887** [comment]
@karthik-tarento Please mention the sunbird release version in the tickets going forward. It would help us to narrow down on issue faster. Also, please share the logs.

**Reply by Jayaprakash8887** [reply]
"post-publish-processor" job does following 3 things and all are currently mandatory processes. There is no confiugration setup to control these process indivudally in **jobs**. You can certainly customize 'PostPublishEventRouter' for adding necessary configurations:
1. Process Shallow Copy Content - verifies if the published collection has any shallow copy collections and triggers publish events for them.
2. Processes Batch creation for "Course"
3. DIAL code linking for a published Course (invlolves DIAL code generation)

cc: @AmiableAnil @vinukumar-vs
